types/react-dom/index.d.ts

```diff
@@ -68,3 +68,11 @@ export function unstable_renderSubtreeIntoContainer<P>(
     element: ReactElement<P>,
     container: Element,
     callback?: (component?: Component<P, ComponentState> | Element) => any): Component<P, ComponentState> | Element | void;
+
+// Portals
+
+export type ReactPortalElement = ReactElement<{}>;
```

Comment:
I'm still of the opinion that this should not be a `ReactElement`. The implementation is at https://github.com/facebook/react/blob/master/src/renderers/shared/fiber/isomorphic/ReactPortal.js#L30-L37, and you can see that the only overlapping property is `key`.

If need be, I think you should define a new interface for `ReactPortal` in `react/index.d.ts`, and include it in the `ReactNode` type there.